In [1]:
import pandas as pd
import numpy as np

In [3]:
table = pd.read_html("https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M")

In [32]:
data = table[0]

In [33]:
data.head()

Postal Code           Borough               Neighborhood
0         M1A      Not assigned               Not assigned
1         M2A      Not assigned               Not assigned
2         M3A        North York                  Parkwoods
3         M4A        North York           Victoria Village
4         M5A  Downtown Toronto  Regent Park, Harbourfront

In [34]:
data.shape

(180, 3)

In [35]:
len(data['Postal Code'].unique())

180

In [36]:
data = data[data.Borough != 'Not assigned']

In [37]:
data

Postal Code           Borough  \
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
5           M6A        North York   
6           M7A  Downtown Toronto   
..          ...               ...   
160         M8X         Etobicoke   
165         M4Y  Downtown Toronto   
168         M7Y      East Toronto   
169         M8Y         Etobicoke   
178         M8Z         Etobicoke   

                                          Neighborhood  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
5                     Lawrence Manor, Lawrence Heights  
6          Queen's Park, Ontario Provincial Government  
..                                                 ...  
160      The Kingsway, Montgomery Road, Old Mill North  
165                               Church and Wellesley  
168  Business reply mail Processing Centre, South C...  
169  Old Mill South, King's Mill Park, Sunnylea, Hu...  
178  Mimico NW, The Queensway West, South of Bloor,...  

[103 rows x 3 columns]

In [38]:
data.shape

(103, 3)

In [31]:
len(data['Postal Code'].unique())

103

In [52]:
(data.Neighborhood=='Not assigned').value_counts()

False    103
Name: Neighborhood, dtype: int64

the number of rows and number of unique element in Postal code, both are 103. Hence, we can say that all the postal codes are uniqe. 

In [29]:
# we can us below code to get elements
# X = data.groupby(['Postal Code']).agg(lambda col: ','.join(col))